recipes/full_finetune_single_device.py

```diff
@@ -313,13 +313,15 @@ def train(self) -> None:
         # zero out the gradients before starting training
         self._optimizer.zero_grad()
 
+        monitor = utils.perf_utils.TunePerfMonitor()
```

Comment:
nit: I would just import this directly instead of using the full filepath here
```suggestion
        monitor = TunePerfMonitor()
```